### ISSUE TITLE
Gradient color stops are not working correctly for webkit

### ISSUE THREAD

**Issue by chrisdrackett** [Compass/compass#118]
Using a position for a color stop creates the correct result in firefox, but not webkit. Example below:

The following in compass:

+linear-gradient(color-stops(#fff,#ddd .1))

yields the following css:

```
    background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #ffffff), color-stop(100%, #dddddd));
    background-image: -moz-linear-gradient(top, #ffffff 0%, #dddddd 10%);
```

notice how the webkit gradient isn't using the 10% value while the firefox code is.

**Comment by imathis** [comment]
Good catch, right now the easiest way to achieve your desired effect is to do this
+linear-gradient(color-stops(#fff, #ddd, .1, #ddd))

It's less beautiful, but it'll work in the mean time while we get this fixed.

**Comment by chrisdrackett** [comment]
another case that breaks even worse:

+linear-gradient(color-stops(#e0e0e0 1,#a8a8a8 0.95,#b3b3b3 0.0))

  background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(Infinity%, #e0e0e0), color-stop(Infinity%, #a8a8a8), color-stop(NaN%, #b3b3b3));
  background-image: -moz-linear-gradient(top, #e0e0e0 100%, #a8a8a8 95%, #b3b3b3 0%);

**Comment by imathis** [comment]
So this doesn't throw an error?

Try again, but instead of using 1, use a percentage, either a decimal or a number that is less than 100 followed by a percent symbol. Using '1' isn't meant to work and should throw an error.

**Comment by chrisdrackett** [comment]
ah, sorry, that fixed it.

**Comment by chriseppstein** [comment]
Raise an error if color stops are specified in the wrong order.
Closed by 910083b2f736311cab22051736006c8bee81653c.

**Comment by chriseppstein** [comment]
The issue here is that the gradient stops were specified in the wrong order.

**Comment by draebb** [comment]
This issue is still exists. `+linear-gradient(color-stops(white, blue 33%, black 67%))` (from document) yields:

```
background-image: -webkit-gradient(linear, 0% 0%, 0% 100%, color-stop(0%, #ffffff), color-stop(49.254%, #0000ff), color-stop(100%, #000000));
background-image: -moz-linear-gradient(top, #ffffff 0%, #0000ff 33%, #000000 67%);
background-image: linear-gradient(top, #ffffff 0%, #0000ff 33%, #000000 67%);
```

It seems caused by `grad_color_stops` that unnecessary converts units which are already relative.

**Comment by chriseppstein** [comment]
Bug fix: In webkit, when the last gradient color stop was a percent less than 100%, the
  intermediate values were being scaled to that value. Closed by a40fce8af1920959a8d7e95d5e5ce7088b2558e9.
